### PR TITLE
Add `private_swiftui_state` SwiftLint rule and resolve the warnings 

### LINKED ISSUE #308: Add `private_swiftui_state` SwiftLint rule and resolve the warnings
## Describe the feature you want to improve
Add `private_swiftui_state` SwiftLint rule (notice they're in alphabetical order) and resolve the warnings

### PR DESCRIPTION
**Pull request by aking618**
# What it Does
* Closes #308 
* This PR adds the `private_swiftui_state` rule to the apps SwiftLint configuration and resolve any warnings regarding the rule.

# How I Tested
1. Build the application
2. Ensure no warnings are generated due to non-private SwiftUI state.
image

# Notes
* N/A

# Screenshot
* N/A

### CODE REVIEW

File: Basic-Car-Maintenance/Shared/Settings/Views/ChooseAppIconView.swift [hunk 1]
```diff
@@ -18,6 +18,11 @@ struct ChooseAppIconView: View {
         GridItem(.flexible(minimum: 30, maximum: 300)),
         GridItem(.flexible(minimum: 30, maximum: 300))
     ]
+    
+    init(viewModel: ChooseAppIconViewModel) {
+        self.viewModel = viewModel
+    }
+
```
Comment (mikaelacaron) [inline Basic-Car-Maintenance/Shared/Settings/Views/ChooseAppIconView.swift]: why is this needed to initialize the object? when it's private and already being created on line `11`

**Review by mikaelacaron — CHANGES_REQUESTED** [review]
Great work!! I have a question that needs to be resolved please. If you have any questions, you can comment directly on that thread, otherwise click the resolve button once you've addressed my comment. Once you fix everything, please click the re-review button

also I've made changes to the .swiftlint file, so you'll need to update your feature branch from `dev`
I've written how to do this in the DocC documentation, you can also find it here: https://mikaelacaron.github.io/Basic-Car-Maintenance/documentation/basic_car_maintenance/

File: Basic-Car-Maintenance/Shared/Settings/Views/ChooseAppIconView.swift (hunk 1 above)
Comment (aking618) [reply Basic-Car-Maintenance/Shared/Settings/Views/ChooseAppIconView.swift]: I added the init to resolve the error an `SettingsView.swift:169` where a `ChooseAppIconView` is instantiated.
Changing the state to private made the synthesized `init` private . 

I can remove the `init` and use the default initializer:
```swift
ChooseAppIconView()
```
instead of
```swift
ChooseAppIconView(viewModel: ChooseAppIconViewModel())
```

Comment (mikaelacaron) [reply Basic-Car-Maintenance/Shared/Settings/Views/ChooseAppIconView.swift]: Yes do that, because looking at the code the `ChooseAppIconViewModel` doesn't need to be initialized in the `SettingsView`, `SettingsView` doesn't use it

and note how it was written in the `SettingsView`, adding the initializer isn't correct, and shouldn't be possible that way, because you've made the variable private, so initializing it outside the struct doesn't make sense

Comment (mikaelacaron) [reply Basic-Car-Maintenance/Shared/Settings/Views/ChooseAppIconView.swift]: Once you do that, resolve this comment and click re-review please! thanks!

Re-review button

**Review by mikaelacaron — APPROVED** [review]
Lovely! Thanks for contributing!
Feel free to make suggestions in discussions of features to add or create new issues of any bugs you find!
